### PR TITLE
DATA-1293 - Allowed id attribute in addition to client-uri attribute in MongoDbFactoryParser.

### PR DESCRIPTION
**Pull request by vkhoroshko**
Additionally, the previous approach allowed any attribute to be set if write-concern attribute was set. Fixed that also.

### CODE REVIEW

File: spring-data-mongodb/src/main/java/org/springframework/data/mongodb/config/MongoDbFactoryParser.java [hunk 1]
```diff
@@ -73,7 +74,10 @@ protected AbstractBeanDefinition parseInternal(Element element, ParserContext pa
 		BeanDefinition mongoUri = getMongoUri(element);
 
 		if (mongoUri != null) {
-			if (element.getAttributes().getLength() >= 2 && !element.hasAttribute("write-concern")) {
```
Comment (olivergierke) [inline spring-data-mongodb/src/main/java/org/springframework/data/mongodb/config/MongoDbFactoryParser.java]: Can we rather white-list the allowed attributes?

Comment (vkhoroshkodirectstream) [inline spring-data-mongodb/src/main/java/org/springframework/data/mongodb/config/MongoDbFactoryParser.java]: Done.

Comment (vkhoroshko) [inline spring-data-mongodb/src/main/java/org/springframework/data/mongodb/config/MongoDbFactoryParser.java]: Done.

**Comment by christophstrobl** [comment]
thanks @vkhoroshko! 
Would you (if you haven't done so yet) please sign the [Spring CLA form](https://support.springsource.com/spring_committer_signup) and add your CLA Number as a comment either here or to [DATAMONGO-1293](https://jira.spring.io/browse/DATAMONGO-1293). I can then take it from here.

**Comment by vkhoroshkodirectstream** [comment]
Sure, 

140120150929074128

**Comment by christophstrobl** [comment]
pushed to master and back ported to 1.8.x (gosling). thanks @vkhoroshko !

**Pull request closed without merging**